### PR TITLE
Generic descriptors for ip2 that will ease combining with ip

### PR DESCRIPTION
**Pull request by kgerheiser**
This re-factors ip2 so that its functions take a generic `ip_grid_descriptor` object instead of the actual grib template arrays (from which the grid_descriptors are constructed).

Then, if a function needs to access the contents of the descriptor it calls `select type` which will give you either a `grib1_descriptor` or a `grib2_descriptor` that can be queried for the needed information.

This is a step towards generalizing the library so that it can be combined with NCEPLIBS-ip.

I tried to leave as much of the code intact as possible for to get it into better shape first, before a bigger re-factoring can commence.

Vector and scalar tests pass and all interfaces remain the same with the exception of a required `use ip_mod`

### CODE REVIEW

**Comment by kgerheiser** [comment]
Oops commited my build files

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -15,7 +15,7 @@ if(${CMAKE_Fortran_COMPILER_ID} MATCHES "^(GNU)$" AND ${CMAKE_Fortran_COMPILER_V
   set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -w -fallow-argument-mismatch -fallow-invalid-boz")
 endif()
 
-set(kinds "4" "8" "d")
+set(kinds "d")
```
Comment (GeorgeGayno-NOAA) [inline src/CMakeLists.txt]: Don't forget to set this back to '4', '8', 'd' when you are done testing.

File: src/gdswzd_c.F90 [hunk 2]
```diff
@@ -226,7 +227,10 @@ SUBROUTINE GDSWZD_C(IGDTNUM,IGDTMPL,IGDTLEN,IOPT,NPTS,FILL, &
  REAL(KIND=C_FLOAT), INTENT(OUT)          :: YLON(NPTS),YLAT(NPTS),AREA(NPTS)
 #endif
```
Comment (GeorgeGayno-NOAA) [inline src/gdswzd_c.F90:229]: I often call ip and ip2 only for the gdswzd routine (no interpolation).  In that case, I can declare a 'type(grib1/grib2_descriptor)' variable in my program, then call gdswzd, correct?  That is a public function?

**Review by GeorgeGayno-NOAA — COMMENTED** [review]
I think you are on the right track.

**Comment by edwardhartnett** [comment]
There are lots of whitespace changes. Did you reformat the code?

I have no objection to reformatting to a common indentation style, but we should agree on the style (across all NCEPLIBS) and then enforce it automatically. Also we should convert to it in one operation in which there are no code changes, only automatic reformatting.

**Comment by kgerheiser** [comment]
I use the style that emacs uses. I added modules around the existing code and then indented it.

File: src/gdswzd_c.F90 (hunk 2 above)
Comment (kgerheiser) [reply src/gdswzd_c.F90:229]: I left the external interface the same. It converts to a descriptor internally. Though, for C I think it'll have to be two different functions like `gdswzd_grib2` or something.

**Comment by kgerheiser** [comment]
This is complete and ready to be merged.

The next step will be adding the grib1-specific code and interfaces.

**Pull request closed without merging**